superset/db_engine_specs.py

```diff
@@ -980,19 +981,82 @@ def get_columns(
             result.append(cls._create_column_info(column, column.Column, column_type))
         return result
 
+    @classmethod
+    def _is_column_name_quoted(cls, column_name: str) -> bool:
+        """
+        Check if column name is in quotes
+        :param column_name: column name
+        :return: boolean
+        """
+        return column_name.startswith('"') and column_name.endswith('"')
+
+    @classmethod
+    def _get_fields(cls, cols: List[dict]) -> List[ColumnClause]:
+        """
+        Format column clauses where names are in quotes and labels are specified
+        :param cols: columns
+        :return: column clauses
+        """
+        column_clauses = []
+        dot_regex = r'\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)'
+        for col in cols:
+            # get individual column names
+            col_names = re.split(dot_regex, col['name'])
+            # quote each column name if it is not already quoted
+            for index, col_name in enumerate(col_names):
+                if not cls._is_column_name_quoted(col_name):
+                    col_names[index] = '"{}"'.format(col_name)
+            quoted_col_name = '.'.join(col_names)
+            # create column clause in the format "name"."name" AS "name.name"
+            column_clause = sqla.literal_column(quoted_col_name).label(col['name'])
+            column_clauses.append(column_clause)
+        return column_clauses
+
+    @classmethod
+    def _filter_presto_cols(cls, cols: List[dict]) -> List[dict]:
+        """
+        We want to filter out columns that correspond to array content because you cannot
+        select array content without an index, which will lead to a large and complicated
+        query. We know which columns to skip because cols is a list provided to us in a
```

Comment:
@khtruong, can you explain what you mean with "you cannot select array content without an index, which will lead to a large and complicated query."?

Reply:
Lol, yeah I'm not surprised that it is still confusing. Let me re-phrase it.